From 4421a7b3f874d0a9220aca311a5e1bfa72ed9ab0 Mon Sep 17 00:00:00 2001
From: Alexander Strange <astrange@ithinksw.com>
Date: Sun, 22 Nov 2009 16:33:58 -0500
Subject: [PATCH 4/5] Hardcode results of runtime cpu-detection in dsputil for SSE2/3DNOW.

This isn't the same as fixing --disable-runtimecpudetection in mainline,
since we still need to run it to detect features newer than SSE2.

There are still unused mmx2 and c DSP functions in the binary,
presumably due to missed optimizations.

Results in ~400KB Perian smaller binary.

TODO:
- fix --disable-3dnow to actually disable this in mainline.
---
 libavcodec/x86/dsputil_mmx.c |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/libavcodec/x86/dsputil_mmx.c b/libavcodec/x86/dsputil_mmx.c
index c4202d5..db388cb 100644
--- a/libavcodec/x86/dsputil_mmx.c
+++ b/libavcodec/x86/dsputil_mmx.c
@@ -2497,6 +2497,25 @@ static void float_to_int16_interleave_3dn2(int16_t *dst, const float **src, long
 
 float ff_scalarproduct_float_sse(const float *v1, const float *v2, int order);
 
+#undef FF_MM_MMX
+#undef FF_MM_MMX2
+#undef FF_MM_3DNOW
+#undef FF_MM_3DNOWEXT
+#undef FF_MM_SSE
+#undef FF_MM_SSE2
+#undef CODEC_FLAG_BITEXACT
+
+#define ON 0 || 1
+#define OFF 0
+
+#define FF_MM_MMX ON
+#define FF_MM_MMX2 ON
+#define FF_MM_3DNOW OFF
+#define FF_MM_3DNOWEXT OFF
+#define FF_MM_SSE ON
+#define FF_MM_SSE2 ON
+#define CODEC_FLAG_BITEXACT 0
+
 void dsputil_init_mmx(DSPContext* c, AVCodecContext *avctx)
 {
     mm_flags = mm_support();
-- 
1.7.1.1

